dev-tools/mage/common.go

```diff
@@ -353,6 +353,69 @@ func unzip(sourceFile, destinationDir string) error {
 	return nil
 }
 
+// Tar compress a directory using tar + gzip algorithms
+func Tar(src string, targetFile string) error {
+	fmt.Printf(">> creating TAR file from directory: %s, target: %s\n", src, targetFile)
+
+	f, err := os.Create(targetFile)
+	if err != nil {
+		return fmt.Errorf("error creating tar file: %w", err)
+	}
+	defer f.Close()
+
+	// tar > gzip > file
+	zr := gzip.NewWriter(f)
+	tw := tar.NewWriter(zr)
+
+	// walk through every file in the folder
+	filepath.Walk(src, func(file string, fi os.FileInfo, errFn error) error {
```

Comment:
Shouldn't this have an error check for `if errFn != nil`.

Reply:
Good catch! Sent a fix in https://github.com/elastic/beats/pull/27295/commits/6d165b54c910e2094768fb82b1b47e6a8343d459

Thanks!